### PR TITLE
feat: Dropdownlist options

### PR DESCRIPTION
**Pull request by HeleenSG**
### Changes

Adds new functionality to dropdown lists: 
- Grouping 
- Subtitles on labels

Please note that the html for this element has changed to support the option of grouping. This has been updated in all current instances of this component. 

New HTML-structure:
```
<div aria-expanded="false" class="dropdown-list">
   <ul>
      <li>
          <a href=""><span class="icon ti-[icon-name]" aria-hidden="true"></span>Scan for objects</a>
      </li>
    </ul>
</div>
``` 

**Grouping**
To use grouping within a dropdown list. Group related elements in ul's within ".dropdownlist".

```
<div aria-expanded="false" class="dropdown-list">
   <ul>
      <li>
          <a href=""><span class="icon ti-[icon-name]" aria-hidden="true"></span>Scan for objects</a>
      </li>
    </ul>

   <ul>
      <li>
          <a href=""><span class="icon ti-[icon-name]" aria-hidden="true"></span>Scan for objects</a>
      </li>
    </ul>
</div>
```

**Icons**
To add icons within a label use one of the following options: 

If the icon is decorative (e.g an edit icon when the text within the button (or link) clearly states that the function of the button is to edit). The following can be used: 
`<button><span class="icon ti-[icon-name]" aria-hidden="true"></span>Edit</button>`

or:
`<a href=""><span class="icon ti-[icon-name]" aria-hidden="true"></span>Scan for objects</a>`

**Subtitle**
When adding a subtitle, group the icon and link text in a div. Add <span class="nota-bene">Lorem ipsum</span> after. 

### Issue link
Closes: https://github.com/minvws/nl-kat-coordination/issues/3249

### Demo

Dropdown list with icons, grouping, and subtitle as an example. 
Screenshot 2024-08-09 at 15 23 24

### QA notes

- Check if all dropdown menu's that have been changed (html changes) still work as intended.
- Check if styling has been added according to the designs. https://www.figma.com/design/MDbpc9K0qjyaoRGAGiDX6t/%F0%9F%8E%A8-KAT-%7C-Design-System-%7C-Components?node-id=86-15112&m=dev (component library > dropdown)

### Code Checklist

- [ ] All the commits in this PR are properly PGP-signed and verified.
- [ ] This PR only contains functionality relevant to the issue.
- [ ] I have written unit tests for the changes or fixes I made.
- [ ] I have checked the documentation and made changes where necessary.
- [ ] I have performed a self-review of my code and refactored it to the best of my abilities.

- [ ] Tickets have been created for newly discovered issues.
- [ ] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [ ] I have informed others of any required `.env` changes files if required and changed the `.env-dist` accordingly.
- [ ] I have included comments in the code to elaborate on what is not self-evident from the code itself, including references to issues and discussions online, or implicit behavior of an interface.

---

## Checklist for code reviewers:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---

## Checklist for QA:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

File: rocky/package.json [hunk 1]
```diff
@@ -16,5 +16,6 @@
   },
   "browserslist": [
     "last 1 Chrome version"
-  ]
+  ],
+  "packageManager": "yarn@1.22.22+sha512.a6b2f7906b721bba3d67d4aff083df04dad64c399707841b7acf00f6b133b7ac24255f2652fa22ae3534329dc6180534e98d17432037ff6fd140556e2bb3137e"
```
Comment (underdarknl) [inline rocky/package.json:20]: Was this an intended change?

File: rocky/reports/report_types/aggregate_organisation_report/introduction.html [hunk 2]
```diff
@@ -8,18 +8,20 @@ <h1>{{ report_name }}</h1>
...
-                           target="_blank"
-                           rel="noopener noreferrer">{% translate "Download JSON" %}</a>
-                    </li>
-                </ul>
+                <div id="export-add" aria-expanded="false" class="dropdown-list">
+                    <ul>
+                        <li>
+                            <a href="{{ report_download_pdf_url }}"
+                               target="_blank"
+                               rel="noopener noreferrer">{% translate "Download PDF" %}</a>
```
Comment (underdarknl) [inline rocky/reports/report_types/aggregate_organisation_report/introduction.html:16]: rel="noopener noreferrer" are not needed on internal / local links. They are not problematic either.

File: rocky/package.json (hunk 1 above)
Comment (HeleenSG) [reply rocky/package.json:20]: Ah no, it's an automatic one...

File: rocky/reports/report_types/aggregate_organisation_report/introduction.html (hunk 2 above)
Comment (HeleenSG) [reply rocky/reports/report_types/aggregate_organisation_report/introduction.html:16]: Ah yes, I will remove it as it's not needed.

**Review by underdarknl — APPROVED** [review]

**Comment by stephanie0x00** [comment]
## Checklist for QA:

- [x] I have checked out this branch, and successfully ran a fresh `make reset`.
- [x] I confirmed that there are no unintended functional regressions in this branch:
  - [x] I have managed to pass the onboarding flow
  - [x] Objects and Findings are created properly
  - [x] Tasks are created and completed properly
- [x] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.
- [x] I checked the logs for errors and/or warnings and made issues where necessary

### What works:
Looks good! The icons really improve the user experience. 

### What doesn't work:
n/a

### Bug or feature?:
n/a

**Review by underdarknl — APPROVED** [review]

**Pull request merged**